server/src/main/java/org/tctalent/server/request/candidate/SearchCandidateRequest.java

```diff
@@ -152,6 +166,226 @@ public SearchCandidateRequest() {
         super(Sort.Direction.DESC, new String[]{"id"});
     }
 
+    /**
+     * Extracts the database query SQL corresponding to the given search request.
+     *
+     * @param nativeQuery True if native database SQL is required, otherwise JPQL is returned
+     * @return String containing the SQL or JPQL
+     */
+    public String extractSQL(boolean nativeQuery,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 47. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 190-202, lines 213-221, lines 226-240, lines 326-366, lines 368-379 into separate methods.